lib/inertia_rails/renderer.rb

```diff
@@ -93,16 +105,30 @@ def page
       }
     end
 
-    def deep_transform_values(hash, proc)
-      return proc.call(hash) unless hash.is_a? Hash
+    def deep_transform_values(hash, &block)
+      return block.call(hash) unless hash.is_a? Hash
 
-      hash.transform_values {|value| deep_transform_values(value, proc)}
+      hash.transform_values {|value| deep_transform_values(value, &block)}
+    end
+
+    def drop_partial_except_keys(hash)
+      partial_except_keys.each do |key|
+        parts = key.to_s.split('.').map(&:to_sym)
+        *initial_keys, last_key = parts
```

Comment:
this support dot notation, yea? so like `except: ['users.some_association.expensive_thing_to_compute']`?

do the other adapters support that? I don't see it documented anywhere.

either way, if we support it for `:except`, should we also support it for `:only`? (genuine question)

Reply:
> do the other adapters support that? I don't see it documented anywhere.
yup, Laravel adapter supports that

> if we support it for :except, should we also support it for :only?
Laravel plugin does not 😄 To be honest the whole `except` + `only` thing is poorly documented, we probably should help the core team make it happen.

Reply:
Actually they reverted back dot notation for `only`: https://github.com/inertiajs/inertia-laravel/commit/4ce06f6fafe9405e5c58b38638de33e0240c53e8

Reply:
Oh nice find!

Reply:
Ok, i dug into this a bit. TL;DR: I don't think those issues will affect our implementation.

Best I can tell (caveat: using my brain's PHP interpreter) is that the original implementation of Laravel's `share` shallow merged all data. However, dot notation allowed you to deep merge dot-notated shared data. Not sure whether that was intentional or a happy accident since I don't think the feature was ever documented. When dot notation was added to `only` and `except`, it changed the execution order and changed the behavior such that _all_ shared data was a shallow merge. So, people relying on the deep merge behavior had breaking changes. (There was another reported but with lazy props that I didn't fully diagnose... looks to be something related to array manipulation quirks in PHP).

InertiaRails 1) has explicit patterns for deep merging shared data and 2) doesn't support dot notation in shared data anyway, so we should be totally fine.

I don't see a reason why we couldn't support dot notation for `only` as well. Do you think it's worth adding? Seems like it'd be nice if some prop had multiple expensive child props and you only wanted to grab one of them.

Reply:
Yup, let's try adding support for ’only’ in a separate PR 👍